python/lsst/analysis/drp/_treecorrConfig.py

```diff
@@ -0,0 +1,351 @@
+    correlation function calculations. The default values set for the fields
+    are identical to the default values set in v4.2 of `treecorr`. The
+    parameters that are excluded in this class are
+    'verbose', 'log_file', 'output_dots', 'rng' and 'pairwise' (deprecated).
+    For details about these options, see the documentation for `treecorr`:
+    https://rmjarvis.github.io/TreeCorr/_build/html/correlation2.html
+
+    A separate config class is used instead
+    of constructing a `~lsst.pex.config.DictField` so that mixed types can be
+    supported and the config can be validated at the beginning of the
+    execution. The ``toDict`` provides a minimal dictionary that override only
+    the default values and excludes the key-values pairs when the item is None.
+
+    Notes
+    -----
+    This is intended to be used in CalcRhoStatistics class.
+    """
+
+    nbins = Field(
+        doc=(
+            "How many bins to use. "
+            "(Exactly three of nbins, bin_size, min_sep, max_sep "
+            "are required. If nbins is not given, it will be "
+            "calculated from the values of the other three, "
+            "rounding up to the next highest integer. "
+            "In this case, bin_size will be readjusted to account "
+            "for this rounding up."
+        ),
+        dtype=int,
+        optional=True,
+        check=lambda x: x > 0,
+    )
+
+    bin_size = Field(
+        doc=(
+            "The width of the bins in log(separation). "
+            "Exactly three of nbins, bin_size, min_sep, max_sep are required. "
+            "If bin_size is not given, it will be calculated from the values "
+            "of the other three."
+        ),
+        dtype=float,
+        optional=True,
+    )
+
+    min_sep = Field(
+        doc=(
+            "The minimum separation in units of sep_units, if relevant. "
+            "Exactly three of nbins, bin_size, min_sep, max_sep are required. "
+            "If min_sep is not given, it will be calculated from the values "
+            "of the other three."
+        ),
+        dtype=float,
+        optional=True,
+    )
+
+    max_sep = Field(
+        doc=(
+            "The maximum separation in units of sep_units, if relevant. "
+            "Exactly three of nbins, bin_size, min_sep, max_sep are required. "
+            "If max_sep is not given, it will be calculated from the values "
+            "of the other three."
+        ),
+        dtype=float,
+        optional=True,
+    )
+
+    sep_units = ChoiceField(
+        doc=(
+            "The units to use for the separation values, given as a string. "
+            "This includes both min_sep and max_sep above, as well as the "
+            "units of the output distance values."
+        ),
+        default="radians",
+        dtype=str,
+        optional=True,
+        allowed={
+            units: units for units in ["arcmin", "arc", "degrees", "hours", "radians"]
```

Comment:
I tried that first, but `allowed` had to be a dictionary where the key are the allowed options and the value being it's documentation.

Reply:
Please use standard unit strings. "hours" is not understood by Astropy and should be "h" or "hour" (and in fact the plural forms are not allowed). What unit is `arc`?

Reply:
That was supposed to be `arcsecond`. Good catch Tim. I intended this to be some sort of a thin wrapper around treecorr config (only because our DictField doesn't accept values of different types). I could implement a mapper that converts standard units to treecorr units, but I'd rather fix it (let you fix it) in treecorr or deal with a mapper later on. At the moment, this config is of limited use beyond making rho statistics.

Reply:
Looks like `treecorr` should accept standard units (despite the `TreeCorr.Catalog` docstring).  See: https://github.com/rmjarvis/TreeCorr/blob/86afbdca38aac6325bc95cdc101b771c8ebb02f3/treecorr/config.py#L88-L104.  Maybe the cleanest resolution is to make a map so either the docstring units or the standard units will work.

Reply:
Ah, we can stick with standard units since it only looks for the beginning of the string.